src/api/attachments.tsx

```diff
@@ -24,3 +30,43 @@ export const usePostAttachmentMetadata = (): UseMutationResult<
       postAttachmentMetadata(attachmentMetadata),
   });
 };
+
+export const getAttachment = async (id: string): Promise<AttachmentPostMetadataResponse> => {
+  return storageApi.get(`/attachments/${id}`).then((response) => {
+    return response.data;
+  });
+};
+
+export const useGetAttachment = (
+  id: string
+): UseQueryResult<AttachmentPostMetadataResponse, AxiosError> => {
+  return useQuery({
+    queryKey: ['Attachment', id],
+    queryFn: () => getAttachment(id),
+  });
+};
+
+const getAttachments = async (
+  entityId: string
+): Promise<AttachmentPostMetadataResponse[]> => {
+  const queryParams = new URLSearchParams();
+  queryParams.append('entity_id', entityId);
+  queryParams.append('modified_time', '2024-01-02T13:10:10.000+00:00')
+  queryParams.append('created_time', '2024-01-01T12:00:00.000+00:00')
+
+  return storageApi
+    .get(`/attachments`, {
+      params: queryParams,
+    })
+    .then((response) => response.data);
+};
+
+export const useGetAttachments = (
+  entityId?: string
+): UseQueryResult<AttachmentPostMetadataResponse[], AxiosError> => {
+  return useQuery({
+    queryKey: ['Attachments', entityId],
```

Comment:
I think the usePostAttachmentMetadata should have something like
```typescript
    onSuccess: () => {
      queryClient.invalidateQueries({
        queryKey: ['Attachments'],
      });
    },
```
To ensure this will re-fetch (I am not quite sure what the caching behaviour is at this point so may not effect anything now but could if we add caching back in - the older queries have this still).

Reply:
Done what i think you mean.

Reply:
Yes and that fixed it so that it now updates the table when a new attachment is uploaded as its invalidating the query causing it to fetch the updated list.